core-java-modules/core-java-lang-oop-patterns/src/test/java/com/baeldung/returnfirstnonnull/ReturnFirstNonNullLazyEvaluateUnitTest.java

```diff
@@ -0,0 +1,69 @@
+package returnfirstnonull;
+
+import java.util.Objects;
+import java.util.Optional;
+import java.util.function.Supplier;
+import java.util.stream.Stream;
+
+import org.apache.commons.lang3.ObjectUtils;
+
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+public class ReturnFirstNonNullLazyEvaluateUnitTest {
+
+    private static Object METHOD_B_OBJECT = new Object();
+
+    private Object methodA() { return null; }
+
+    private Object methodB() { return METHOD_B_OBJECT; }
+
+    private Object methodC() { return null; }
```

Comment:
It might be better if `methodC()` returned another non null value, to prove that we are actually returning the first one.

Reply:
Changed as suggested